components/fxa-client/src/send_tab.rs

```diff
@@ -64,7 +66,46 @@ impl FirefoxAccount {
                 }
             };
         let encrypted_payload: EncryptedSendTabPayload = serde_json::from_value(payload)?;
-        let payload = encrypted_payload.decrypt(&send_tab_key)?;
-        Ok(IncomingDeviceCommand::TabReceived { sender, payload })
+        match encrypted_payload.decrypt(&send_tab_key) {
+            Ok(payload) => Ok(IncomingDeviceCommand::TabReceived { sender, payload }),
+            Err(e) => {
+                log::error!("Could not decrypt Send Tab payload. Diagnosing then resetting the Send Tab keys.");
+                match self.diagnose_remote_keys(send_tab_key) {
+                    Ok(_) => log::error!("Could not find the cause of the Send Tab keys issue."),
+                    Err(e) => log::error!("{}", e),
+                };
+                // Reset the Send Tab keys.
+                self.state.commands_data.remove(send_tab::COMMAND_NAME);
+                self.reregister_current_capabilities()?;
```

Comment:
If this network call fails, we'll be left unable to receive any tabs until app restart, which will call `ensure_capabilities` to be called and repair the situation. That seems OK from a "we already couldn't receive tabs anyway" perspective.